x-pack/plugins/event_log/server/es/init.ts

```diff
@@ -35,6 +38,122 @@ class EsInitializationSteps {
     this.esContext = esContext;
   }
 
+  async setExistingIndexTemplatesToHidden() {
+    let indexTemplates: estypes.IndicesGetTemplateResponse = {};
+    try {
+      // look up existing index templates and update index.hidden to true if that
+      // setting is currently false or undefined
+
+      // since we are updating to the new index template API and converting new event log
+      // indices to hidden in the same PR, we only need to use the legacy template API to
+      // look for and update existing event log indices.
+      indexTemplates = await this.esContext.esAdapter.getExistingLegacyIndexTemplates(
+        this.esContext.esNames.indexPattern
+      );
+    } catch (err) {
+      // errors when trying to get existing index templates
+      // should not block the rest of initialization, log the error and move on
+      this.esContext.logger.error(`error getting existing index templates - ${err.message}`);
+    }
+
+    Object.keys(indexTemplates).forEach(async (indexTemplateName: string) => {
+      try {
+        const hidden: string | boolean = indexTemplates[indexTemplateName]?.settings?.index?.hidden;
+        // Check to see if this index template is hidden
+        if (hidden !== true && hidden !== 'true') {
+          this.esContext.logger.debug(
+            `setting existing "${indexTemplateName}" index template to hidden.`
+          );
+
+          await this.esContext.esAdapter.setLegacyIndexTemplateToHidden(
+            indexTemplateName,
+            indexTemplates[indexTemplateName]
+          );
+        }
+      } catch (err) {
+        // errors when trying to update existing index templates to hidden
+        // should not block the rest of initialization, log the error and move on
+        this.esContext.logger.error(
+          `error setting existing "${indexTemplateName}" index template to hidden - ${err.message}`
+        );
+      }
+    });
+  }
+
+  async setExistingIndicesToHidden() {
+    let indices: estypes.IndicesGetSettingsResponse = {};
+    try {
+      // look up existing indices and update index.hidden to true if that
+      // setting is currently false or undefined
+      indices = await this.esContext.esAdapter.getExistingIndices(
+        this.esContext.esNames.indexPattern
+      );
+    } catch (err) {
+      // errors when trying to get existing indices
+      // should not block the rest of initialization, log the error and move on
+      this.esContext.logger.error(`error getting existing indices - ${err.message}`);
+    }
+
+    Object.keys(indices).forEach(async (indexName: string) => {
```

Comment:
nit: it seems like we could have done the "find old indices" search with `expand_wildcards=open`, so it wouldn't return already hidden indices, so wouldn't have had to make this check.  Feels safer to do it this way though :-)